### PR TITLE
Support cache versioning

### PR DESCRIPTION
**Pull request by doits**
## What is the current behavior?

There is no way to use a cache version when deriving the cache key of a record. It currenctly simply uses `record.cache_key`. If [cache versioning](https://github.com/rails/rails/pull/29092) is enabled, the cache key will never change, even if the record got updated.

## What is the new behavior?

Cache versioning can be enabled with:

```rb
cache_options cache_versioning: true
```

It will automatically pass `version: record.cache_version` when fetching the cache.

## Questions

- Should the default automatically be inferred by checking `Rails.application.config.active_record.cache_versioning`?
- Is it OK to pass `version: nil` to `Rails.cache.fetch` if cache versioning is disabled or should the call omit the `version:` in that case? (maybe it could brake older Rails apps?)

## Checklist

Please make sure the following requirements are complete:

- [x] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been reviewed and added / updated if needed (for bug fixes /
  features) --> will be done if this has a chance to be merged
- [ ] All automated checks pass (CI/CD)

### CODE REVIEW

**Comment by doits** [comment]
You're right about the tests, I think the existing tests (and the new ones I added) are very hard to follow and test too much at once - though this is a different matter, so I just tried to follow the convention. Besides this, afaik existing caching tests do not test internals but the outcome when the public API is called (which for this point is a good style and leads to stable tests).

Current tests simply check that the cached version is returned if version is the same and uncached if version changed, regardless of how the internals work.

So there could be tests that mock `Rails.cache` and check for the correct call of `fetch` (e.g. what `version:` is passed), though I usually try to avoid to test such internals, because the next refactoring will break this test (even if the output and public API of the serializer stay the same). Though I could add them if you think they are needed in this case.

**Comment by stas** [comment]
@doits, as per #51, I do believe that the caching support should be pluggable (similar to how it is done in rails), ex: 
```
class MySerializer
  include FastJsonApi::ObjectSerializer

  cache_options(store: ActiveSupport::Cache::MemoryStore.new, ...)
  cache_options(store: Rails.cache, ...)
end
```

Right now, it's just magical 🧙‍♂️ haha.
And I'd like us to move away from mocking things as well.

**Comment by stas** [comment]
Closed in favor of #52
